activesupport/lib/active_support/json/encoding.rb

```diff
@@ -149,25 +149,6 @@ def stringify(jsonified)
       if defined?(::JSON::Coder) && Gem::Version.new(::JSON::VERSION) >= Gem::Version.new("2.15")
         class JSONGemCoderEncoder # :nodoc:
           JSON_NATIVE_TYPES = [Hash, Array, Float, String, Symbol, Integer, NilClass, TrueClass, FalseClass, ::JSON::Fragment].freeze
-          CODER = ::JSON::Coder.new do |value, is_key|
```

Comment:
Ideally the fix should be done on `JSON::Coder` to allow one global instance to reset its state when encountering an error

but in the meantime for rails users on version `2.15` of the json gem i see two solutions :

1- Either delete/comment this entire if branch until it's fixed on the `json`
2- Or initialiaze a new `JSON::Coder` instance everytime we try to encode/dump